x/stakeibc/keeper/keeper.go

```diff
@@ -254,6 +254,31 @@ func (k Keeper) GetICATimeoutNanos(ctx sdk.Context, epochType string) (uint64, e
 
 // safety check: ensure the redemption rate is NOT below our min safety threshold && NOT above our max safety threshold on host zone
 func (k Keeper) IsRedemptionRateWithinSafetyBounds(ctx sdk.Context, zone types.HostZone) (bool, error) {
+	// Get the wide bounds
+	minSafetyThreshold, maxSafetyThreshold := k.GetOuterSafetyBounds(ctx, zone)
+
+	redemptionRate := zone.RedemptionRate
+
+	if redemptionRate.LT(minSafetyThreshold) || redemptionRate.GT(maxSafetyThreshold) {
+		errMsg := fmt.Sprintf("IsRedemptionRateWithinSafetyBounds check failed %s is outside safety bounds [%s, %s]", redemptionRate.String(), minSafetyThreshold.String(), maxSafetyThreshold.String())
+		k.Logger(ctx).Error(errMsg)
+		return false, errorsmod.Wrapf(types.ErrRedemptionRateOutsideSafetyBounds, errMsg)
+	}
+
+	// Verify the redemption rate is within the tight safety bounds
+	// The tight safety bounds should always be within the safety bounds, but
+	// the redundancy above is cheap.
+	if redemptionRate.LT(zone.MinTightRedemptionRate) || redemptionRate.GT(zone.MaxTightRedemptionRate) {
+		errMsg := fmt.Sprintf("IsRedemptionRateWithinSafetyBounds check failed %s is outside tight safety bounds [%s, %s]", redemptionRate.String(), minSafetyThreshold.String(), maxSafetyThreshold.String())
+		k.Logger(ctx).Error(errMsg)
+		return false, errorsmod.Wrapf(types.ErrRedemptionRateOutsideSafetyBounds, errMsg)
+	}
+
+	return true, nil
+}
+
+func (k Keeper) GetOuterSafetyBounds(ctx sdk.Context, zone types.HostZone) (sdk.Dec, sdk.Dec) {
```

Comment:
should we add this for inner as well?

Reply:
I was thinking we didn't need it because we have the zone object with the attributes easily accessible

Although it might make sense to add this function with a bit of validation (e.g. check that the attributes exist, if not default to the outer bounds). If the inner bounds are outside of the outer bounds, could also return the outer bounds.

Wdyt?

Reply:
yeah my thinking was just to coalesce the values if they're uninitialized (like the outer function does). I think defaulting to outer makes sense!

Reply:
added 9a98d4d3d